include/exadg/operators/generalized_laplace_operator.h

```diff
@@ -0,0 +1,744 @@
+  }
+
+  /**
+   * Same as above, but takes in as input the product of the coefficients, the
+   * gradient, and the the face normal vector. This version is for example
+   * useful for boundary integrals, where boundary conditions may prescribe the
+   * value of the this product.
+   */
+  inline DEAL_II_ALWAYS_INLINE //
+    ValueType
+    calculate_value_flux(ValueType const &       coeff_times_gradient_times_normal_m,
+                         ValueType const &       coeff_times_gradient_times_normal_p,
+                         ValueType const &       value_m,
+                         ValueType const &       value_p,
+                         Vector const &          normal,
+                         CoefficientType const & coefficient)
+  {
+    ValueType const    jump_value  = value_m - value_p;
+    GradientType const jump_tensor = outer_product(jump_value, normal);
+
+    ValueType const average_coeff_times_gradient_times_normal =
+      0.5 * (coeff_times_gradient_times_normal_m + coeff_times_gradient_times_normal_p);
+
+    return average_coeff_times_gradient_times_normal -
+           ValueType{coeff_mult(coefficient, (tau * jump_tensor)) * normal};
+  }
+
+  /**
+   * Calculates the interior penalty parameter `tau` for the discretization.
+   */
+  void
+  calculate_penalty_parameter(dealii::MatrixFree<dim, Number> const & matrix_free,
+                              unsigned int const                      dof_index)
+  {
+    IP::calculate_penalty_parameter<dim, Number>(penalty_parameters, matrix_free, dof_index);
+  }
+
+  Coefficients &
+  get_coefficients()
+  {
+    return coefficients;
+  }
+
+  /**
+   * Returns the coefficient for the requested (@p cell index, @p q point index)
+   * pair.
+   *
+   * If the coefficient is not variable, returns the constant coefficient.
+   */
+  CoefficientType
+  get_coefficient_cell(unsigned int const cell, unsigned int const q) const
+  {
+    return std::visit(VarCoeffUtils::GetCoefficientCell<CoefficientType>{cell, q}, coefficients);
+  }
+
+  /**
+   * Returns the coefficient for the requested (@p face index, @p q point index)
+   * pair.
+   *
+   * If the coefficient is not variable, returns the constant coefficient.
+   */
+  CoefficientType
+  get_coefficient_face(unsigned int const face, unsigned int const q) const
+  {
+    return std::visit(VarCoeffUtils::GetCoefficientFace<CoefficientType>{face, q}, coefficients);
+  }
+
+  /**
+   * Returns the coefficient for the requested (@p face index (cell-based), @p q
+   * point index) pair.
+   *
+   * If the coefficient is not variable, returns the constant coefficient.
+   */
+  CoefficientType
+  get_coefficient_face_cell_based(unsigned int const cell_based_face, unsigned int const q) const
+  {
+    return std::visit(VarCoeffUtils::GetCoefficientFaceCellBased<CoefficientType>{cell_based_face,
+                                                                                  q},
+                      coefficients);
+  }
```

Comment:
To be honest, I do not see the advantages of the variant/visitor design patter here. In my opinion, one needs to write more lines of code, and the code is more difficult to understand than a very simple if-else logic. Since we are here in the generalized Laplace operator, there is no concern right now that this if-else logic needs to be duplicated at multiple places. And even in such a case, one could also introduce free functions in the future, but only once we need this complexity. So where is the advantage of the present design? Defining two functions `operator()` with the same name is essentially a complicated way of implementing an if-else logic.